code/modules/mob/dead/observer/observer.dm

```diff
@@ -381,11 +387,17 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	mind.current.client.init_verbs()
 	return TRUE
 
-/mob/dead/observer/proc/stay_dead()
+/mob/dead/observer/verb/stay_dead()
+	set name = "Do Not Resuscitate"
+
 	if(!can_reenter_corpse)
 		to_chat(usr, span_warning("You're already stuck out of your body!"))
 		return FALSE
 
+	var/response = tgui_alert(usr, "Are you sure you want to prevent (almost) all means of resuscitation? This cannot be undone.", "Are you sure you want to stay dead?", list("DNR","Save Me"))
+	if(response != "DNR")
+		return
```

Comment:
I'd rather you make this a separate proc that calls stay_dead rather than make the ghost menu tgui open a separate tgui to confirm.